api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/Enhancement.java

```diff
@@ -72,20 +72,25 @@
      * parameter of any member of the type, or as a meta-annotation on any annotation
      * that is considered by these rules.
      * <p>
-     * Defaults to the {@linkplain BeanDefiningAnnotations set of bean defining annotations}.
+     * If empty, the set of <em>expected types</em> is not narrowed down in any way.
+     * If {@code java.lang.Annotation} is present, the set of <em>expected types</em>
+     * is narrowed down to types that use any annotation.
+     * The {@link BeanDefiningAnnotations @BeanDefiningAnnotations} marker type may
+     * be used to narrow down the set of <em>expected types</em> to types that use
+     * any bean defining annotation.
      * <p>
-     * If empty, or if {@code java.lang.Annotation} is present, all annotations are used.
-     * That is, the set of <em>expected types</em> is narrowed down to the set of types
-     * that use any annotation.
+     * Defaults to an empty array, so that the set of <em>expected types</em> is not
+     * narrowed down in any way.
      *
      * @return types of annotations that must be present on the <em>expected types</em>
      */
-    Class<? extends Annotation>[] withAnnotations() default BeanDefiningAnnotations.class;
+    Class<? extends Annotation>[] withAnnotations() default {};
 
     /**
-     * Marker annotation type that represents set of bean defining annotations after
-     * the {@link Discovery @Discovery} phase is finished. That is, it includes custom
-     * normal scope annotations as well as custom stereotypes.
+     * Marker annotation type that, for the purpose of {@link Enhancement#withAnnotations()},
+     * represents set of bean defining annotations after the {@link Discovery @Discovery}
+     * phase is finished. That is, it includes custom normal scope annotations as well as
+     * custom stereotypes.
```

Comment:
@Ladicek I have been thinking about this some more and I am not sure if `BeanDefiningAnnotations`  makes much sense with latest change (i.e. if it is justifiable to have that coded into spec).
Setting aside that I am not sure how well can we filter out classes with just custom stereotypes (which I will try to figure tomorrow), in what case would you want to filter for these classes? With annotated mode, this basically translates to "all classes". The only ones which are potentially omitted are those added by build compatible extensions that that don't have any annotation on them. That, to me, seems like a weird filter but maybe I am just not seeing the use case.

Reply:
The idea was to indicate document that in annotated mode only classes with at least a bean definition annotation are processed. Having said that there is no technical reason for needing the annotation itself and maybe it is just documentation.

Reply:
I had to think about this for a while, but @manovotn is right. During type discovery, we discovery classes with a BDA and classes added via extension in `@Discovery`. Classes added via extension automatically get `@Dependent` if they don't have a BDA already. So saying "only process classes that were discovered during type discovery and have a BDA" is basically equivalent to "only process classes that were discovered during type discovery".